CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -1958,6 +1958,60 @@ + (NSArray*)getNotifications {
     return notifications;
 }
 
+#pragma mark - Retrieving notifications based on the flag remote/local
++ (NSArray*)getNotifications:(BOOL)combineWithApi {
+    if (combineWithApi) {
+        NSString *combinedURL = [self createDynamicURLForNotification];
+        NSMutableArray* combinedArray = [[self getNotifications] mutableCopy];
+        NSMutableArray *remoteNotifications = [self getRemoteNotifications:combinedURL];
+        return [combinedArray arrayByAddingObjectsFromArray:remoteNotifications];
+    } else {
+        return [self getNotifications];
+    }
+}
+
+#pragma mark - Creating URL based on the topic dialogue and append the topicId's as a query parameter.
++ (NSString *)createDynamicURLForNotification {
+    NSString *baseURL = [NSString stringWithFormat:@"https://api.cleverpush.com/channel/%@/received-notifications?", channelId];
+    if ([self hasSubscriptionTopics]) {
+        NSMutableArray* dynamicQueryParameters = [self urlQueryParameter];
+        NSString* appendableQueryParameters = [dynamicQueryParameters componentsJoinedByString:@""];
+        NSString *concatenatedURL = [NSString stringWithFormat:@"%@%@", baseURL, appendableQueryParameters];
+        return concatenatedURL;
+    } else {
+        return baseURL;
+    }
+}
+
+#pragma mark - Appending the topicId's as a query parameter.
++ (NSMutableArray*)urlQueryParameter {
+    NSMutableArray* subscriptionTopics = [self getSubscriptionTopics];
+    NSMutableArray* dynamicQueryParameter = [NSMutableArray new];
+    [subscriptionTopics enumerateObjectsUsingBlock: ^(id topic, NSUInteger index, BOOL *stop) {
+        NSString *queryParameter = [NSString stringWithFormat: @"topics[]=%@&", topic];
+        [dynamicQueryParameter addObject:queryParameter];
+    }];
+    return dynamicQueryParameter;
+}
+
+#pragma mark - Get the Notifications based on the topic dialog Id's.
++ (NSMutableArray *)getRemoteNotifications:(NSString*)url {
+    __block NSMutableArray* notiications = nil;
+    dispatch_semaphore_t sema = dispatch_semaphore_create(0);
```

Comment:
Please do not work with semaphores here. The `getNotifications` method can have a callback block if needed.